srml/session/src/lib.rs

```diff
@@ -550,13 +560,33 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Disable the validator of index `i`.
-	pub fn disable_index(i: usize) {
-		T::SessionHandler::on_disabled(i);
+	///
+	/// Returns `true` if this causes a third of validators to be already disabled.
+	pub fn disable_index(i: usize) -> bool {
+		let (fire_event, third_reached) = DisabledValidators::mutate(|disabled| {
+			let i = i as u32;
+			if let Err(index) = disabled.binary_search(&i) {
+				let count = <Validators<T>>::decode_len().unwrap_or(0);
+				disabled.insert(index, i);
+				(true, disabled.len() >= (count + 2) / 3)
```

Comment:
What if this overflows?

Reply:
I think in many places in the runtime we assume _count of stuff < u32_. Some examples: 
- number of validators
- number of councilors.
- number of election rounds in council. 

Some strong reasons exist to claim all of them will not exceed u32 in the foreseeable future. But if we want to change them (apply your recommendation), I would suggest doing that for all of them and basically sweeping the runtime for any `as` of normal `ops::{*}` operations that could overflow (I don't think it is necessary tbh)

Reply:
Might be worthwhile to store the _emergency threshold_ of the consensus as a config here? Given that I might --in the future-- want to use this module with a different consensus.

Reply:
I don't think we should expect a >2G `ValidatorSet`, even if we do I think we will run into other issues much earlier (we deserialize & allocate the entire vec in other parts of the code).